staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -162,6 +162,12 @@ func (r *proxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	}
 
 	// we need to wrap the roundtripper in another roundtripper which will apply the front proxy headers
+	//
+	// Question(caesarxuchao): if the req contains upgrade
```

Comment:
I don't understand why aggregator needs an spdy.Roundtripper. I think we can just create a regular upgrade-aware round tripper here. If that's true, we can simplify this code a lot but that will also be a bigger change.

The current PR is a more targeted fix. Let's test it out first.